### PR TITLE
feat: add ContainerResource type for HPA cpu/memory scaler

### LINKED ISSUE #1378: Allow cpu/memory scaler to target a specific container in the pod
### Use-Case
One limitation with Kubernetes' default hpa implementation, is that scaling happens based on the "sum" of all containers in the targeted pod. This means that sidecar metrics affect the scaling, and thus sidecar cpu/memory requests/limits have to be very carefully managed in order to achieve the desired scaling dynamics. Especially in cases where sidecar cpu/memory resources are tiny compared to those of the "main" app container, tuning this correctly is often very difficult. The kubernetes metrics-server actually presents per-container metrics, it's just that the default hpa doesnt have a way of selecting which individual container to track metrics for.

Since keda implements its own cpu/memory scaler, it would be _so_ awesome if a new optional field `container` was added, which allowed using only the metrics of a specific container (ignoring other containers in the pod) for scaling.

### PR DESCRIPTION
**Pull request by surki**
Context: https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#container-resource-metrics

Added feature to target to a specific container in cpu/memory scalers

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs/pull/876
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #1378

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 - **Prometheus Scaler:** Add ignoreNullValues to return error when prometheus return null in values ([#3065](https://github.com/kedacore/keda/issues/3065))
 - **Selenium Grid Scaler:** Edge active sessions not being properly counted ([#2709](https://github.com/kedacore/keda/issues/2709))
 - **Selenium Grid Scaler:** Max Sessions implementation issue ([#3061](https://github.com/kedacore/keda/issues/3061))
+- **CPU/Memory Scaler:** Added support for targeting specific container in a pod ([#1378](https://github.com/kedacore/keda/issues/1378))
```
Comment (zroubalik) [inline CHANGELOG.md:61]: Please sort this alphabetically.

**Review by zroubalik — COMMENTED** [review]
What happens if we apply this resource on k8s < 1.20?

File: CHANGELOG.md (hunk 1 above)
Comment (surki) [reply CHANGELOG.md]: Done

**Comment by surki** [comment]
> What happens if we apply this resource on k8s < 1.20?

When we run it in k8s < 1.20 or with 1.20+ with the `HPAContainerMetrics` feature disabled, HPA object creation will fail, which I suppose the KEDA will propagate back to the user.

**Review by zroubalik — APPROVED** [review]

**Comment by tomkerkhove** [comment]
> > What happens if we apply this resource on k8s < 1.20?
> 
> When we run it in k8s < 1.20 or with 1.20+ with the `HPAContainerMetrics` feature disabled, HPA object creation will fail, which I suppose the KEDA will propagate back to the user.

I would, however, improve this so that there is no HPA failure and we log an error + block ScaledObject/ScaledJob creation/edit to fail fast and notify the user.

**Comment by zroubalik** [comment]
> > > What happens if we apply this resource on k8s < 1.20?
> > 
> > 
> > When we run it in k8s < 1.20 or with 1.20+ with the `HPAContainerMetrics` feature disabled, HPA object creation will fail, which I suppose the KEDA will propagate back to the user.
> 
> I would, however, improve this so that there is no HPA failure and we log an error + block ScaledObject/ScaledJob creation/edit to fail fast and notify the user.

That happens already, failure in HPA -> failure in ScaledObject/Job.

**Comment by aaron-sysdig** [comment]
hey @zroubalik was this change ever released? I dont see it in main? https://github.com/surki/keda/commits/main/pkg/scalers/cpu_memory_scaler.go

**Comment by zroubalik** [comment]
Yes, it was. You need to check https://github.com/kedacore/keda/ not a fork

**Pull request merged**